### PR TITLE
condense XPath for unnecessary_lambda_linter()

### PR DESCRIPTION
**Pull request by MichaelChirico**
This will make the actual XPath more readable; encountered in pursuing #1567 

before:

```xpath
//SYMBOL_FUNCTION_CALL[ text() = 'lapply' or text() = 'sapply' or text() = 'vapply' or text() = 'apply' or text() = 'tapply' or text() = 'rapply' or text() = 'eapply' or text() = 'dendrapply' or text() = 'mapply' or text() = 'by' or text() = 'outer' or text() = 'mclapply' or text() = 'mcmapply' or text() = 'parApply' or text() = 'parCapply' or text() = 'parLapply' or text() = 'parLapplyLB' or text() = 'parRapply' or text() = 'parSapply' or text() = 'parSapplyLB' or text() = 'pvec' or text() = 'map' or text() = 'walk' or text() = 'map_raw' or text() = 'map_lgl' or text() = 'map_int' or text() = 'map_dbl' or text() = 'map_chr' or text() = 'map_vec' or text() = 'map_df' or text() = 'map_dfr' or text() = 'map_dfc' ]
  /parent::expr
  /following-sibling::expr[
    FUNCTION
    and count(SYMBOL_FORMALS) = 1
    and expr/OP-LEFT-PAREN/following-sibling::expr[1][not(preceding-sibling::*[1][self::EQ_SUB])]/SYMBOL
    and SYMBOL_FORMALS = expr/OP-LEFT-PAREN/following-sibling::expr[1]/SYMBOL
    and not(SYMBOL_FORMALS = expr/OP-LEFT-PAREN/following-sibling::expr[position() > 1]//SYMBOL)
  ]|//SYMBOL_FUNCTION_CALL[ text() = 'lapply' or text() = 'sapply' or text() = 'vapply' or text() = 'apply' or text() = 'tapply' or text() = 'rapply' or text() = 'eapply' or text() = 'dendrapply' or text() = 'mapply' or text() = 'by' or text() = 'outer' or text() = 'mclapply' or text() = 'mcmapply' or text() = 'parApply' or text() = 'parCapply' or text() = 'parLapply' or text() = 'parLapplyLB' or text() = 'parRapply' or text() = 'parSapply' or text() = 'parSapplyLB' or text() = 'pvec' or text() = 'map' or text() = 'walk' or text() = 'map_raw' or text() = 'map_lgl' or text() = 'map_int' or text() = 'map_dbl' or text() = 'map_chr' or text() = 'map_vec' or text() = 'map_df' or text() = 'map_dfr' or text() = 'map_dfc' ]
  /parent::expr
  /following-sibling::expr[
    FUNCTION
    and count(SYMBOL_FORMALS) = 1
    and expr[OP-LEFT-BRACE and count(expr) = 1]/expr[1]/OP-LEFT-PAREN/following-sibling::expr[1][not(preceding-sibling::*[1][self::EQ_SUB])]/SYMBOL
    and SYMBOL_FORMALS = expr[OP-LEFT-BRACE and count(expr) = 1]/expr[1]/OP-LEFT-PAREN/following-sibling::expr[1]/SYMBOL
    and not(SYMBOL_FORMALS = expr[OP-LEFT-BRACE and count(expr) = 1]/expr[1]/OP-LEFT-PAREN/following-sibling::expr[position() > 1]//SYMBOL)
  ]
```

after:

```xpath
//SYMBOL_FUNCTION_CALL[ text() = 'lapply' or text() = 'sapply' or text() = 'vapply' or text() = 'apply' or text() = 'tapply' or text() = 'rapply' or text() = 'eapply' or text() = 'dendrapply' or text() = 'mapply' or text() = 'by' or text() = 'outer' or text() = 'mclapply' or text() = 'mcmapply' or text() = 'parApply' or text() = 'parCapply' or text() = 'parLapply' or text() = 'parLapplyLB' or text() = 'parRapply' or text() = 'parSapply' or text() = 'parSapplyLB' or text() = 'pvec' or text() = 'map' or text() = 'walk' or text() = 'map_raw' or text() = 'map_lgl' or text() = 'map_int' or text() = 'map_dbl' or text() = 'map_chr' or text() = 'map_vec' or text() = 'map_df' or text() = 'map_dfr' or text() = 'map_dfc' ]
    /parent::expr
    /following-sibling::expr[(FUNCTION or OP-LAMBDA) and count(SYMBOL_FORMALS) = 1]
    /expr[last()][
      count(.//SYMBOL[self::* = preceding::SYMBOL_FORMALS[1]]) = 1
      and count(.//SYMBOL_FUNCTION_CALL) = 1
      and preceding-sibling::SYMBOL_FORMALS =
        //expr[
          position() = 2
          and preceding-sibling::expr/SYMBOL_FUNCTION_CALL
          and not(preceding-sibling::*[1][self::EQ_SUB])
        ]/SYMBOL
    ]
    /parent::expr
```

### CODE REVIEW

File: tests/testthat/test-unnecessary_lambda_linter.R [hunk 1]
```diff
@@ -34,28 +34,36 @@ test_that("unnecessary_lambda_linter skips allowed usages", {
     NULL,
     linter
   )
+
+  # This _could_ be lapply(x, `%in%`, tbl), but don't force infix into lambda
+  expect_lint("lapply(x, function(xi) xi %in% tbl)", NULL, linter)
+  # This one could not
+  expect_lint("lapply(x, function(xi) tbl %in% xi)", NULL, linter)
```
Comment (AshesITR) [inline tests/testthat/test-unnecessary_lambda_linter.R:41]: This one could also, using a named arg, be ``lapply(x, `%in%`, x = tbl)``.

File: R/unnecessary_lambda_linter.R [hunk 2]
```diff
@@ -60,22 +60,26 @@ unnecessary_lambda_linter <- function() {
...
     /parent::expr
     /following-sibling::expr[
       FUNCTION
       and count(SYMBOL_FORMALS) = 1
-      and {paren_path}/OP-LEFT-PAREN/following-sibling::expr[1][not(preceding-sibling::*[1][self::EQ_SUB])]/SYMBOL
-      and SYMBOL_FORMALS = {paren_path}/OP-LEFT-PAREN/following-sibling::expr[1]/SYMBOL
-      and not(SYMBOL_FORMALS = {paren_path}/OP-LEFT-PAREN/following-sibling::expr[position() > 1]//SYMBOL)
+      and (
+        (
+          { glue(symbol_tests_fmt, call_expr_path = 'expr') }
```
Comment (AshesITR) [inline R/unnecessary_lambda_linter.R:76]: Nested glue calls reduce readability IMO.
Please factor them out into local variables.

File: tests/testthat/test-unnecessary_lambda_linter.R (hunk 1 above)
Comment (MichaelChirico) [reply tests/testthat/test-unnecessary_lambda_linter.R:41]: even harder to read, but indeed :)

**Review by AshesITR — APPROVED** [review]
I noticed we have no tests for metadata (location).
Feel free to add some or merge as-is.

**Comment by MichaelChirico** [comment]
#2221 let's hand that throughout the suite & try to prevent regression as well

**Pull request merged**